components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1850,12 +1949,24 @@ void RewardsServiceImpl::GetAutoContribute(
   bat_ledger_->GetAutoContribute(std::move(callback));
 }
 
-void RewardsServiceImpl::SetAutoContribute(bool enabled) const {
+void RewardsServiceImpl::SetAutoContribute(bool enabled) {
   if (!Connected()) {
     return;
   }
 
   bat_ledger_->SetAutoContribute(enabled);
+  auto_contributions_enabled_ = enabled;
+
+  // Record stats.
+  DCHECK(profile_->GetPrefs()->GetBoolean(prefs::kBraveRewardsEnabled));
```

Comment:
why is this check needed?

Reply:
if I understand it will DCHECK crash if rewards are not enabled?

Reply:
yep, it assumes this. Is it a wrong assumption? In other words, is there a situation when `SetAutoContribute` can be called when this pref is false?